tests/test_collection.py

```diff
@@ -349,6 +349,76 @@ def test_no_filter(self):
         self.assertRaises(IOError, self.c.filter)
 
 
+class PropertiesNumberFormattingTest(unittest.TestCase):
+
+    def setUp(self):
+        self.tempdir = tempfile.mkdtemp()
+        self.filename = os.path.join(self.tempdir, "properties_number_formatting_test")
+
+    def tearDown(self):
+        shutil.rmtree(self.tempdir)
+
+    def _write_collection(self, driver):
+        writing_schema = {
+            'geometry': 'Point',
+            'properties': [('integer', 'int'), ('one_digit', 'float:15.1')]
+        }
+
+        with fiona.open(
+                self.filename,
+                "w",
+                driver=driver,
+                schema=writing_schema,
+                crs='epsg:4326',
+                encoding='utf-8'
+        ) as c:
+            c.writerecords([
+                {
+                    'geometry': {'type': 'Point', 'coordinates': (0.0, 0.1)},
+                    'properties': {'integer': 12.22, 'one_digit': 12.22}
+                },
+                {
+                    'geometry': {'type': 'Point', 'coordinates': (0.0, 0.2)},
+                    'properties': {'integer': 12.88, 'one_digit': 12.88}
+                }
+            ])
+
+    def test_shape_file_properties_number_formatting(self):
+        driver = "ESRI Shapefile"
+        self._write_collection(driver)
+
+        with fiona.open(self.filename, driver=driver, encoding='utf-8') as c:
+            self.assertEqual(len(c), 2)
+
+            rf1, rf2 = list(c)
+
+            # integers are truncated
+            self.assertEqual(rf1['properties']['integer'], 12)
+            self.assertEqual(rf2['properties']['integer'], 12)
+
+            # floats are rounded
```

Comment:
@galak75 would you be willing to make the float assertions into a separate test function? The truncation and rounding effects have different causes.